src/java.base/share/classes/java/util/zip/ZipOutputStream.java

```diff
@@ -80,6 +82,33 @@ public XEntry(ZipEntry entry, long offset) {
 
     private final ZipCoder zc;
 
+    /* Cache SOURCE_DATE_EPOCH if set for reproducible Zip content */
+    private static final long sourceDateEpochMillis = getSourceDateEpochMillis();
+
+    @SuppressWarnings("removal")
+    private static long getSourceDateEpochMillis() {
+        long sourceDateEpoch = AccessController.doPrivileged(
+            new PrivilegedAction<Long> () {
+                public Long run() {
+                    long value;
+                    String env = System.getenv("SOURCE_DATE_EPOCH");
+                    if (env != null) {
+                        try {
+                            value = Long.parseLong(env);
+                            // SOURCE_DATE_EPOCH is in seconds
+                            value *= 1000;
+                        } catch(NumberFormatException e) {
+                            value = -1;
+                        }
+                    } else {
+                        value = -1;
+                    }
+                    return new Long(value);
```

Comment:
value can be returned here directly which will use auto-boxing. (new Long(long) is deprecated).